index.js

```diff
@@ -45,8 +46,23 @@ class vehicle {
             }
 
             if (result.status == 200) {
-                this.token = result.data.access_token
-                resolve(result.data.access_token)
+                var tokenOptions = {
+                    method: 'PUT',
+                    baseURL: tokenUrl,
+                    url: '/api/oauth2/v1/token',
+                    headers: Object.fromEntries(fordHeaders),
+                    data: { code: result.data.access_token }
+                }
+                console.log(tokenOptions)
+                try {
+                    var tokenResult = await request(tokenOptions)
+                    console.log(tokenResult.data)
+                } catch (err)  {
+                    console.log(err)
+                    reject(err.result.status)
+                }
+                this.token = tokenResult.data.access_token
```

Comment:
Maybe check for `tokenResult.status == 200`?

Reply:
I second this suggestion.

Reply:
This could be done like so:

```suggestion
                if (tokenResult.status != 200) {
                    console.log(tokenResult)
                    reject(tokenResult.status)
                    return
                }
                this.token = tokenResult.data.access_token
```